pypgstac/pypgstac/load.py

```diff
@@ -137,15 +148,18 @@ def read_json(file: Union[Path, str, Iterator[Any]] = "stdin") -> Iterable:
                 yield orjson.loads(line)
 
 
-@dataclass
```

Comment:
Remove dataclass; this is more appropriate as a normal Python object. IMO dataclasses should have light functionality and really be kept to data containers with public members (sort of like a struct). The dataclass requirement for immutable defaults was forcing the need to make this private attribute optional, which triggered the change from a dataclass to a standard object.